### PR TITLE
Add wordcount and line count filtering to address #89

### PR DESCRIPTION
**Pull request by evanrichter**
# word and line count filtering

addresses #89 

## Branching checklist
- [x] There is an issue associated with your PR (bug, feature, etc.. if not, create one)
- [x] Your PR description references the associated issue (i.e. fixes #123)
- [x] Code is in its own branch
- [x] Branch name is related to the PR contents
- [x] PR targets master

## Static analysis checks
- [x] All rust files are formatted using `cargo fmt`
- [x] All `clippy` checks pass when running `cargo clippy --all-targets --all-features -- -D warnings -A clippy::unnecessary_unwrap`
- [x] All existing tests pass

## Documentation
- [x] New code is documented using [doc comments](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html)
- [x] Documentation about your PR is included in the README, as needed

## Additional Tests
- [x] New code is unit tested
- [x] New code is integration tested, as needed
- [x] New tests pass

### CODE REVIEW

File: src/heuristics.rs [hunk 1]
```diff
@@ -186,11 +186,15 @@ async fn make_wildcard_request(
                 let ferox_response = FeroxResponse::from(response, false).await;
```
Comment (epi052) [inline src/heuristics.rs:186]:
```suggestion
                let ferox_response = FeroxResponse::from(response, true).await;
```

File: src/heuristics.rs [hunk 2]
```diff
@@ -186,11 +186,15 @@ async fn make_wildcard_request(
...
                 let url_len = get_url_path_length(&ferox_response.url());
                 let content_len = ferox_response.content_length();
+                let content_words = ferox_response.word_count();
+                let content_lines = ferox_response.line_count();
 
                 if !CONFIGURATION.quiet && !should_filter_response(&ferox_response) {
                     let msg = format!(
-                        "{} {:>10} Got {} for {} (url length: {})\n",
+                        "{} {:>8} {:>8} {:>8} Got {} for {} (url length: {})\n",
```
Comment (epi052) [inline src/heuristics.rs:194]: On lines 92 and 113 of this file, the final message looks like this:

`WLD       3526 Wildcard response is static; auto-filtering 3526 responses; toggle this behavior by using --dont-filter`

They should probably have the same column formatting as the others. (i can't add a comment on the exact lines for w/e reason)

File: src/reporter.rs [hunk 3]
```diff
@@ -104,8 +104,10 @@ async fn spawn_terminal_reporter(
                 format!(
                     // example output
                     // 200       3280 https://localhost.com/FAQ
-                    "{} {:>10} {}\n",
+                    "{} {:>8} {:>8} {:>8} {}\n",
```
Comment (epi052) [inline src/reporter.rs:107]: Current output:
`200       83      333     3526 https://....`

Proposed:
`200       83l      333w     3526c https://....`

I'm open to other thoughts here, tho the short options for lines, words, and bytes according to the wc command seem pretty intuitive

Comment (epi052) [reply src/reporter.rs:107]: also, the heuristics output should be update to match

**Review by epi052 — CHANGES_REQUESTED** [review]
The PR looks great, thank you!  

Other than what's already noted in the review, lines 479 and 523 in scanner.rs should be updated to read the body of the incoming response.  

Thank you again!

**Comment by evanrichter** [comment]
those changes are incorporated now!

**Comment by evanrichter** [comment]
as for the clippy [weirdness](https://rust-lang.github.io/rust-clippy/master/#deref_addrof), I confirm that it does not happen in current nightly with versions of stuff:

```
❯ cargo clippy -V
clippy 0.0.212 (f5230fb 2020-11-16)

❯ cargo -V       
cargo 1.49.0-nightly (2af662e22 2020-11-12)

❯ rustc -V
rustc 1.50.0-nightly (f5230fbf7 2020-11-16)
```

The latest commit simply appeases clippy stable, and should be removed in a couple months or so 🤷

**Review by epi052 — APPROVED** [review]

**Comment by epi052** [comment]
Changes look good!  i'll keep an eye out for when stable clippy catches up with the current nightly.  Thank you for your help!

**Pull request merged**